pkg/cmd/clients.go

```diff
@@ -126,77 +130,150 @@ Run the "mobile get clients" command from this tool to get the client ID.`,
 			if err != nil {
 				return errors.Wrap(err, "failed to get namespace")
 			}
-			app := &v1alpha1.MobileClient{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "MobileClient",
-					APIVersion: "mobile.k8s.io/v1alpha1",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      map[string]string{},
-					Annotations: map[string]string{},
-				},
-				Spec: v1alpha1.MobileClientSpec{Name: name, ApiKey: appKey, ClientType: clientType, AppIdentifier: appIdentifier},
-			}
-			switch app.Spec.ClientType {
-			case "android":
-				app.Annotations["icon"] = "fa fa-android"
-				break
-			case "iOS":
-				app.Annotations["icon"] = "fa fa-apple"
-				break
-			case "cordova":
-				app.Annotations["icon"] = "font-icon icon-cordova"
-				break
-			case "xamarin":
-				app.Annotations["icon"] = "font-icon icon-xamarin"
-				break
+
+			if appIdentifier == ""{
+				return errors.New("failed validation while creating new mobile client")
+			}
+
+			var apbName string
+			switch clientType {
+			case "android", "iOS", "cordova", "xamarin":
+				apbName = clientType+"-app"
 			default:
 				return errors.New("Unknown client type")
+
+			}
+
+			clientId := name + "-" + clientType
+			client, err := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Get(clientId, metav1.GetOptions{})
+			if client.ObjectMeta.UID != ""{
+				return errors.New("App with this name already exist for this client type")
 			}
-			app.Name = name + "-" + strings.ToLower(app.Spec.ClientType)
-			if err := input.ValidateMobileClient(app); err != nil {
-				return errors.Wrap(err, "failed validation while creating new mobile client")
+			if err != nil {
+				return errors.Wrap(err, "failed to check if application name exists")
 			}
-			createdClient, err := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Create(app)
+
+			//Get available provision parameters from the cluster service plan
+			clusterServiceClass, err := findServiceClassByName(cc.scClient, apbName)
 			if err != nil {
+				return errors.Wrap(err, "failed to find ServiceClass by name")
+			}
+
+			validServiceName := clusterServiceClass.Spec.ExternalName
+			extMeta := clusterServiceClass.Spec.ExternalMetadata.Raw
+			var extServiceClass ExternalServiceMetaData
+			if err := json.Unmarshal(extMeta, &extServiceClass); err != nil {
+				return errors.Wrap(err, "failed to read ClusterServiceClass")
+			}
+
+			si := buildServiceInstance(namespace, validServiceName+"-", clientId + "-apb-" + "params", *clusterServiceClass)
+
+			if _, err := cc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Create(&si); err != nil {
 				return errors.Wrap(err, "failed to create mobile client")
 			}
-			outType := outputType(cmd.Flags())
-			if err := cc.Out.Render("create"+cmd.Name(), outType, createdClient); err != nil {
-				return errors.Wrap(err, fmt.Sprintf(output.FailedToOutPutInFormat, "mobile client", outType))
+			fmt.Println("Creating Mobile Client")
```

Comment:
Should this be here?

Reply:
it takes a while to provision, so I followed the pattern from creating service instance and log it out to let the user know what is happening